File: emarsys/src/main/java/com/emarsys/config/ConfigInternal.java
package com.emarsys.config;

import com.emarsys.core.api.experimental.FlipperFeature;
import com.emarsys.core.api.result.CompletionListener;

import java.util.List;

public interface ConfigInternal {

    void setContactFieldId(int contactFieldId);

    int getContactFieldId();

    void changeApplicationCode(String applicationCode, CompletionListener completionListener);

    String getApplicationCode();

    void changeMerchantId(String merchantId);

    String getMerchantId();

    List<FlipperFeature> getExperimentalFeatures();
}

File: emarsys/src/main/java/com/emarsys/config/ConfigInternal.kt
package com.emarsys.config

import com.emarsys.core.api.notification.NotificationSettings
import com.emarsys.core.api.result.CompletionListener

interface ConfigInternal {

    val contactFieldId: Int?

    val applicationCode: String?

    val merchantId: String?

    val clientId: String

    val language: String

    val notificationSettings: NotificationSettings

    val isAutomaticPushSendingEnabled: Boolean

    val sdkVersion: String

    fun changeApplicationCode(applicationCode: String?, completionListener: CompletionListener?)

    fun changeMerchantId(merchantId: String?)

    fun refreshRemoteConfig(completionListener: CompletionListener?)

    fun resetRemoteConfig()
}
